schemas/core/data_type.go

```diff
@@ -139,14 +139,15 @@ func NewDataType(t string) (DataType, error) {
 	case ComplexType:
 		return &Complex{}, nil
 	}
-	return nil, &dataTypeError{"Invalid type"}
+	return nil, &DataTypeError{"Invalid type"}
```

Comment:
@alelb Message should be more descriptive (ie. including the invalid type name)

Reply:
Pass `t` handling message formatting within the `DataTypeError`.

Evaluate also to rename `DataTypeError`, into `InvalidDataTypeError`.